src/dict.c

```diff
@@ -541,8 +541,8 @@ void *dictFetchValue(dict *d, const void *key) {
  * the fingerprint again when the iterator is released.
  * If the two fingerprints are different it means that the user of the iterator
  * performed forbidden operations against the dictionary while iterating. */
-long long dictFingerprint(dict *d) {
-    long long integers[6], hash = 0;
```

Comment:
hash(signed variable) is shifted beyond its size(UB) or becomes negative and shifted again (UB)